port/unix/omrsysinfo.c

```diff
@@ -3074,31 +3091,49 @@ retrieveLinuxCgroupMemoryStats(struct OMRPortLibrary *portLibrary, struct OMRCgr
 	cgroupMemInfo->memoryAndSwapUsage = OMRPORT_MEMINFO_NOT_AVAILABLE;
 	cgroupMemInfo->cached = OMRPORT_MEMINFO_NOT_AVAILABLE;
 
-	rc = readCgroupSubsystemFile(portLibrary, OMR_CGROUP_SUBSYSTEM_MEMORY, CGROUP_MEMORY_LIMIT_IN_BYTES_FILE, numItemsToRead, "%lu", &cgroupMemInfo->memoryLimit);
+	if (OMR_ARE_ANY_BITS_SET(PPG_sysinfoControlFlags, OMRPORT_SYSINFO_CGROUP_V1_AVAILABLE)) {
+		memLimitFile = CGROUP_MEMORY_LIMIT_IN_BYTES_FILE;
+		memUsageFile = CGROUP_MEMORY_USAGE_IN_BYTES_FILE;
+		swpLimitFile = CGROUP_MEMORY_SWAP_LIMIT_IN_BYTES_FILE;
+		swpUsageFile = CGROUP_MEMORY_SWAP_USAGE_IN_BYTES_FILE;
+		metricName = CGROUP_MEMORY_STAT_CACHE_METRIC;
+		metricSize = CGROUP_MEMORY_STAT_CACHE_METRIC_SZ;
```

Comment:
`metricName` and `metricSize` are generic. They should be given a meaningful name. `cachedMetricName` and `cachedMetricNameSize`.